### DISCUSSION TITLE
Generating graphs with data that comes in relationships

### DISCUSSION THREAD

**Discussion by arpadgabor** [Budibase/budibase#4863] [Help] [answered]
I have 2 tables in the native db driver, Logs and Projects, and a project can have many logs. I'm trying to generate a graph of hours spent on each project, however the Project field in logs comes in as an object. I would've tried parsing this data with javascript but I can't find a way to do that either. Any ideas? Or is this not supported yet? Even in the views feature this does not seem to be possible.

![image](https://user-images.githubusercontent.com/9270494/157543139-2fc6ec3f-1615-4ebf-9414-13db42dd31d8.png)

**Comment by mjashanks** [comment, marked as answer]
Hey Arpad.

You should create a Formula column on your table, and use that instead. Make sure you create a "Dynamic" formula:

![image](https://user-images.githubusercontent.com/3524181/157634988-2c5f08fc-9a34-4345-9a09-d5878760f1c4.png)

Your binding will look like `{{ Project.0.Name }}` or in JS `return $("Project")[0].Name;`

**Reply by arpadgabor** [reply]
This does seem to make it work somehow, but the graph is not grouped by that column :( using the date groups fine though. Also, is it possible to use dynamic columns inside views? I can see you can filter by a formula column however in the group by dropdown formula columns are not showing up.

**Reply by mjashanks** [reply]
Yes the reason why it's not showing up in views is because it's a dynamic filter - i.e. it runs client side, not server side. Unfortunately it's not possible to use this inside views. There's lots we'd like to add to views, but not sure when we are going to get to this: https://github.com/Budibase/budibase/discussions/2999
